File: potumTest.js
potumTest = TestCase("potumTest");

potumTest.prototype.testFactPostive = function() {
	var actual = fact(5);
	var expected = 120;
	assertEquals(expected, actual);
};

potumTest.prototype.testFactNegative = function() {
	var actual = fact(-5);
	var expected = 1;
	assertEquals(expected, actual);
};

potumTest.prototype.testFactDouble = function() {
	var actual = fact(5.5);
	var expected = 162.421875;
	assertEquals(expected, actual);
};

potumTest.prototype.testFactNull = function() {
	var actual = fact(null);
	var expected = 1;
	assertEquals(expected, actual);
};

potumTest.prototype.testFactString = function() {
	var actual = fact("One");
	var expected = 1;
	assertEquals(expected, actual);
};

potumTest.prototype.testPi = function() {
	var actual = calcpi();
	var expected = Math.PI;
	assertEquals(expected, actual);
};

potumTest.prototype.testPowPositive = function() {
	var result = pow(2,3);
	var expected = Math.pow(2,3);
	assertEquals(expected, result);
};

potumTest.prototype.testPowNullFirst = function() {
	var result = pow(null,3);
	var expected = Math.pow(null,3);
	assertEquals(expected, result);
};

potumTest.prototype.testPowBothNull = function() {
	var result = pow(null,null);
	var expected = Math.pow(null,null);
	assertEquals(expected, result);
};

potumTest.prototype.testPowNullLast = function() {
	var result = pow(-2.2,null);
	var expected = Math.pow(-2.2, null);
	assertEquals(expected, result);
};

potumTest.prototype.testPowString = function() {
	var result = pow("hi","hi");
	var expected = Math.pow("hi","hi");
	assertEquals(expected, result);
};

potumTest.prototype.testPowNegative = function() {
	var result = pow(-2,-2);
	var expected = Math.pow(-2,-2);
	assertEquals(expected, result);
};

potumTest.prototype.testPowBaseNegative = function() {
	var result = pow(-2,2);
	var expected = Math.pow(-2,2);
	assertEquals(expected, result);
};
potumTest.prototype.testPowDouble = function() {
	var result = pow(2.5,2.3);
	var expected = Math.pow(2.5,2.3);
	assertEquals(expected, result);
};

potumTest.prototype.testLSinPostive = function() {
	var x = 2;
	var result = lsin(x) + x;
	var expected = Math.sin(x);
	assertEquals(expected, result);
};

potumTest.prototype.testLSinZero = function() {
	var x = 0;
	var result = lsin(x) + x;
	var expected = lsin(x) + x;
	assertEquals(expected, result);
};

potumTest.prototype.testLSinNull = function() {
	var x = null;
	var result = lsin(x) + x;
	var expected = Math.sin(x);
	assertEquals(expected, result);
};

potumTest.prototype.testLSinString = function() {
	var x = "String";
	var result = lsin(x) + x;
	var expected = "NaNString";
	assertEquals(expected, result);
};

potumTest.prototype.testLSinDouble = function() {
	var x = 1.1;
	var result = lsin(x) + x;
	var expected = Math.sin(x);
	assertEquals(expected, result);
};

potumTest.prototype.testLSinNegative = function() {
	var x = -1;
	var result = lsin(x) + x;
	var expected = Math.sin(x);
	assertEquals(expected, result);
};
potumTest.prototype.testLCosPostive = function() {
	var x = 1;
	var result = lcos(x) + 1;
	var expected = Math.cos(x);
	assertEquals(expected, result);
};

potumTest.prototype.testLCosZero = function() {
	var x = 0;
	var result = lcos(x) + 1;
	var expected = Math.cos(x);
	assertEquals(expected, result);
};

potumTest.prototype.testLCosNull = function() {
	var x = null;
	var result = lcos(x) + 1;
	var expected = Math.cos(x);
	assertEquals(expected, result);
};

potumTest.prototype.testLCosDouble = function() {
	var x = 3.3;
	var result = lcos(x) + 1;
	assertNotNull(result);
	assertNotUndefined(result);
};

potumTest.prototype.testLCosNegative = function() {
	var x = -3;
	var result = lcos(x) + 1;
	assertNotNull(result);
	assertNotUndefined(result);
};
potumTest.prototype.testSolve = function() {
	var a = solve(f, 0, 2 * pi);
	var expected = 2.3098814600100575;
	assertEquals(expected, a);
};

File: potumTest.scala

import scala.math.BigInt.int2bigInt
import org.junit.Test
import junit.framework.Assert.assertEquals

class potumTest extends potum {

  @Test def testFactPositive() {
    val actual: BigInt = po.fact(5);
    val i: BigInt = 120;
    assertEquals(actual, i);

  };

  @Test def testFactNegative() {
    val actual: BigInt = po.fact(-5);
    val i: BigInt = 1;
    assertEquals(actual, i);
  };

  @Test def testFactDouble() {
    val temp: BigInt = BigInt("5");
    val actual = po.fact(temp);
    val expected = 120;
    assertEquals(expected, actual);
  };

  @Test def testFactNull() {
    val t: BigInt = Double.NaN.toInt;
    val actual = po.fact(t);
    val expected = 1;
    assertEquals(expected, actual);
  };

  @Test def testFactString() {
    val temp: BigInt = BigInt("1");
    val actual = po.fact(temp);
    val expected = 1;
    assertEquals(expected, actual);
  };

  @Test def testPi() {
    val actual: Double = po.calcpi();
    val expected: Double = Math.PI;
    assertEquals(expected, actual);
  };

  @Test def testPowPositive() {
    val result: Double = po.pow(2, 3);
    val expected: Double = 8;
    assertEquals(expected, result);
  };
  @Test def testPowNegative() {
    val result: Double = po.pow(-2, -2);
    val expected: Double = 1.0;
    assertEquals(expected, result);
  };

  @Test def testPowNegativeA() {
    val result: Double = po.pow(-2, 2);
    val expected: Double = 4;
    assertEquals(expected, result);
  };
  @Test def testPowNull() {
    val t: Double = Double.NaN;
    val result = po.pow(t, 3);
    val expected = t;
    assertEquals(expected, result);
  };

  

  @Test def testLSinNegative() {
    var x = -2;
    var result = po.lsin(x) + x;
    var expected = Math.sin(x);
    assertEquals(expected, result);
  };
  @Test def testLSinPositive() {
    var x = 2;
    var result = po.lsin(x) + x;
    var expected = 0.9092974268256817;
    assertEquals(expected, result);
  };
  @Test def testLSinZero() {
    var x = 0;
    var result = po.lsin(x) + x;
    var expected = 0.0;
    assertEquals(expected, result);
  };

  @Test def testLSinNull() {
    val t: Double = Double.NaN;
    var x = null;
    var result = po.lsin(t) + x;
    var expected = t;
    assertEquals(expected, result);
  };

  @Test def testLSinDouble() {
    var x = 1.2345;
    var t: Double = x.toDouble;
    var result = po.lsin(t) + x;
    var expected =0.9439833239445112;
    assertEquals(expected, result);
  };
@Test def testLSinNegativeDouble() {
    var x = -5.5;
    var t: Double = x.toDouble;
    var result = po.lsin(t) + x;
    var expected =0.7055405250952838;
    assertEquals(expected, result);
  };

  @Test def testLCosNegative() {
    var x = -1;
    var result = po.lcos(x) + 1;
    var expected = Math.cos(x);
    assertEquals(expected, result);
  };

  @Test def testLCosPositive() {
    var x = 2;
    var result = po.lcos(x) + 1;
    var expected = -0.4161468365471426;
    assertEquals(expected, result);
  };

  @Test def testLCosZero{
    var x=0;
     var result = po.lcos(x) + 1;
    var expected = 1.0;
    assertEquals(expected, result);
    
  }
  @Test def testLCosNull() {
    val t: Double = Double.NaN;
    var result = po.lcos(t) + 1;
    var expected = Math.cos(t);
    assertEquals(expected, result);
  };

  @Test def testLCosDouble() {
    var x = 3.3333;
    var t: Double = x.toDouble;
    var result = po.lcos(t) + 1;
    var expected = -0.9816803564367413;
    assertEquals(expected, result);
  };

  @Test def testSolveAlpha() {
    var a = po.solve(0, 2 * po.pi);
    var expected = 2.3098814600100575;
    assertEquals(expected, a);
  };

  @Test def testSolveLength() {
    var r: Double = 1;
    var expected: Double = 1.1920544934009658;
    var alpha: Double = 2.3098814600100575;
    var actual = 2 * r * (1 - Math.cos(alpha / 2))
    assertEquals(expected, actual);
  };
@Test def testSolveLengthA() {
    var r: Double = 50;
    var expected: Double = 59.60272467004829;
    var alpha: Double = 2.3098814600100575;
    var actual = 2 * r * (1 - Math.cos(alpha / 2))
    assertEquals(expected, actual);
  };
}